lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -521,17 +788,79 @@ class DocumentImpl extends NodeImpl {
     this._modified();
     return this;
   }
-  close() {
+  close(noQueue) {
+    // In some cases like when creating an empty iframe, I want to emit the
+    // events right away to avoid problems if later I asign the property src.
+    if (noQueue) {
+      this.readyState = "complete";
+
+      const ev = this.createEvent("HTMLEvents");
+      const ev2 = this.createEvent("HTMLEvents");
+      ev.initEvent("DOMContentLoaded", true, false);
+      ev2.initEvent("load", false, false);
+
+      this.dispatchEvent(ev);
+      this.dispatchEvent(ev2);
+
+      return;
+    }
     this._queue.resume();
 
-    // Set the readyState to 'complete' once all resources are loaded.
-    // As a side-effect the document's load-event will be dispatched.
-    resourceLoader.enqueue(this, null, function () {
+    const dummyPromise = new Promise(resolve => {
+      resolve();
+    });
+
+    const onDOMContentLoad = () => {
+      const doc = this;
+      function dispatchEvent() {
+        // https://html.spec.whatwg.org/#the-end
+        doc.readyState = "interactive";
+        const ev = doc.createEvent("HTMLEvents");
+        ev.initEvent("DOMContentLoaded", true, false);
+        doc.dispatchEvent(ev);
+      }
+
+      // return Promise.resolve();
+      return new Promise(resolve => {
+        if (!this._deferQueue.tail) {
+          dispatchEvent();
+          return resolve();
+        }
+
+        this._deferQueue.setListener(() => {
+          dispatchEvent();
+          resolve();
+        });
+
+        return this._deferQueue.resume();
+      });
+    };
+
+    const onLoad = () => {
       this.readyState = "complete";
       const ev = this.createEvent("HTMLEvents");
-      ev.initEvent("DOMContentLoaded", false, false);
+
+      ev.initEvent("load", false, false);
+
       this.dispatchEvent(ev);
-    })(null, true);
+
+      return Promise.resolve();
```

Comment:
This will not be necessary if you do the above changes to avoid `.then` testing.

Reply:
This is used in the queue, not in the load you comment above, but I can apply something similar to the queue :)